backends/arm/test/tosautil/tosa_test_utils.py

```diff
@@ -164,6 +165,11 @@ def run_tosa_ref_model(
             params_input[0], params_input[1], inputs
         ):
             data_np = data.detach().numpy()
+
+            if permute_memory_to_nhwc:
+                NHWC_Order = (0, 2, 3, 1)
+                data_np = np.transpose(data_np, NHWC_Order)
```

Comment:
Shouldn't this transpose be done outside this method? I.e. this method takes in whatever tensor and runs the model. If it happens to be NHWC that is ok.

Reply:
agree but i suggest left it here for now due to: 
- we have the `permute_memory_to_nhwc` flag to annotate should it happen or not here as a temporary fix. 
-  this patch will likely be reverted in the long term once Songhao's fix is done. Add a simple check here instead of a new function is a little bit easier to maintain.

Reply:
Fred has a followup patch to improve this as well.